docs/csharp/whats-new/csharp-7.md

```diff
@@ -361,17 +361,20 @@ modified. The local variable has been declared with the `ref` modifier,
 and it will take a `ref` return. You must initialize a `ref` variable when
 it is declared, you cannot split the declaration and the initialization.
 
-The C# language has two other rules that protect you from misusing
+The C# language has three other rules that protect you from misusing
 the `ref` locals and returns:
 
 * You cannot assign a value to a `ref` variable.
     - That disallows statements like `ref int i = sequence.Count();`
 * You cannot return a `ref` to a variable whose lifetime does not extend beyond the execution of the method.
-    - That means you cannot return a reference to a local variable, or similar scope.
+    - That means you cannot return a reference to a local variable or a variable with a similar scope.
+* You cannot return a `ref` to a variable that doesn't exist when the method is ready to set or return the reference.
```

Comment:
Whether or not the variable exists isn't really the issue -- it's that async methods return (almost) immediately before doing any work, so there's no way of knowing whether the value of the variable has been set to its final value or not.